Adding otterscan to erigon docker-compose

mariuspod opened the pull request:
Hey @wmitsuda!
I've glued together otterscan with the recent erigon into the existing `docker-compose.yml` .
Inside `Dockerfile.otterscan` there's a temporary redirect to my fork of otterscan which we can remove soon.
Lmk what you think and if it's useful for you. For me it already is since I had my erigon already running and now I've added otterscan on top of it! :joy:

Still impressed how fast otterscan is :rocket: :rocket: :rocket: 
Thanks for the project :raised_hands:

wmitsuda committed:
First public alpha of otterscan extensions

wmitsuda committed:
Rebase and fixes necessary to run otterscan on top of v2021.07.01 erigon tag

mariuspod committed:
Adding otterscan to the docker-compose setup and linking it with the ots-enabled rpcdaemon. Created a startup script for handling permissions correctly for mounted volumes when starting prometheus and grafana.

mariuspod committed:
Replace --chaindata by --datadir as in upstream repo.

wmitsuda commented:
Hey @mariuspod , since you seem to be a docker heavyuser, could you try the `feature/docker-experiments` branch of otterscan repo?

I managed to glue otterscan + 4bytes + trustwallet in a ~170MB unique docker image. This simplifies a lot the distribution, just serve everything with nginx.

I intend to make it the default way for people to consume otterscan, could you please try it and see if it improves your setup?

mariuspod committed:
Use docker image from docker-experiments feature branch of otterscan.

mariuspod commented:
Sure, this makes a lot of sense!

I've removed my Dockerfile from this repo in favor of the otterscan repo and updated the docker-compose here.
So far the service is behaving like before and is quite usable :joy:

mariuspod committed:
Use the ghcr docker image for otterscan.

wmitsuda commented:
There are now docker-compose templates at: https://github.com/wmitsuda/one-liner-archive

Closing this PR in favor of improve the ones at one-liner-archive repo.
